hugr-core/src/hugr/validate.rs

```diff
@@ -743,6 +747,9 @@ pub enum ValidationError {
     /// There are errors in the extension deltas.
     #[error(transparent)]
     ExtensionError(#[from] ExtensionError),
+    /// A node claims to still be awaiting extension inference. Perhaps it is not acted upon by inference...
```

Comment:
Fair, updated message too.

However, just to note - any Hugr-node that contains TO_BE_INFERRED in its delta is going to either be an error (if its parent doesn't) or result in a TLD that is basically unusable (because no runtime/environment/`run_on_quantum_hw` node/etc. will support such extensions). So we *could* skip both the error and the check. Just a thought....